src/aws/mod.rs

```diff
@@ -156,6 +165,7 @@ pub async fn create_client<T: ClientBuilder>(
 
     // Build the configuration first.
     let mut config_builder = SdkConfig::builder()
+        .credentials_cache(auth.credentials_cache().await?)
         .credentials_provider(auth.credentials_provider(region.clone()).await?)
```

Comment:
Everything below here should be reverted, as the existing code was improvements based on PR feedback